### PR TITLE
#2937 #2952 + hardened URL checks

### PR DESCRIPTION
**Pull request by MikhailArkhipov**
#2952 Output name of the R interpreter when it is not found
#2937 After changing connection URL message is misleading
Harden URL checks
Disable connect button when URL is invalid
Handle Enter better
Add bad URL check in HTTP client it case it still leaks there

### CODE REVIEW

File: src/R/Components/Impl/InteractiveWorkflow/Commands/SessionInformationCommand.cs [hunk 1]
```diff
@@ -43,9 +44,15 @@ public sealed class SessionInformationCommand : IAsyncCommand {
...
+        private void OnRSessionConnected(object sender, EventArgs e) {
             if (_interactiveWorkflow.RSession.IsRemote) {
-                ReplInitComplete().ContinueWith(async (t) => await PrintBrokerInformationAsync(reportTelemetry: true)).DoNotWait();
+                // Host signals 'connected' after R loop was set up (and R banner is printed).
+                // We want to complete printing synchorously here so output does not come
+                // after user started typing in the REPL. But don't wait forever.
+                Task.Run(async () => {
+                    await ReplInitComplete();
+                    await PrintBrokerInformationAsync(reportTelemetry: true);
+                }).Wait(10000);
```
Comment (int19h) [inline src/R/Components/Impl/InteractiveWorkflow/Commands/SessionInformationCommand.cs]: Will this actually block the REPL prompt? And if it does, doesn't that rely on the order of `Connected` events being fired?

File: src/R/Components/Impl/InteractiveWorkflow/Implementation/RInteractiveEvaluator.cs [hunk 2]
```diff
@@ -32,35 +33,49 @@ public sealed class RInteractiveEvaluator : IInteractiveEvaluator {
...
             _console = console;
             _evaluatorRequest = new CountdownDisposable();
-        }
 
-        public void Dispose() {
-            if (CurrentWindow != null) {
-                CurrentWindow.TextView.VisualElement.SizeChanged -= VisualElement_SizeChanged;
-            }
+            _disposableBag
+                .Add(() => Session.Output -= SessionOnOutput)
```
Comment (int19h) [inline src/R/Components/Impl/InteractiveWorkflow/Implementation/RInteractiveEvaluator.cs]: Might as well put all the event deregistration in the single `{}` block that's already there below.

Comment (MikhailArkhipov) [inline src/R/Components/Impl/InteractiveWorkflow/Implementation/RInteractiveEvaluator.cs]: OK. Alex tends to add them separately so I kept the style

File: src/R/Components/Impl/InteractiveWorkflow/Commands/SessionInformationCommand.cs (hunk 1 above)
Comment (MikhailArkhipov) [inline src/R/Components/Impl/InteractiveWorkflow/Commands/SessionInformationCommand.cs]: I looked at R and our host code and the banner is printed in `setupRLoop` which we call (and then offer MRO), then host calls 'connected'. So by the 'connected' time the banner should be in the REPL queue. However, releasing control here will allow other interactions to happen - such as initlialization process and possible even user input.

**Comment by int19h** [comment]
👍

**Pull request merged**